### ISSUE TITLE
S1144 FP: `Deconstruct` method in nested type should be considered

### ISSUE THREAD

**Issue by antoinebj** [SonarSource/sonar-dotnet#2478]
### Description

A false positive exists for rule S1144 when using a deconstructible type.
This is similar to #2126 but the example given in reply failed to reproduce it because the deconstructible type was not nested.

### Repro steps

Run analysis on this code:

```csharp
internal sealed class S1144
{
    public void SomeMethod()
    {
        var (a, b) = new SomeDeconstructibleType();
    }

    private sealed class SomeDeconstructibleType
    {
        public void Deconstruct(out object a, out object b)
        {
            a = b = null;
        }
    }
}
```

### Expected behavior

No violation of S1144 should be detected.

### Actual behavior

A violation of S1144 is detected.

### Known workarounds

Suppression of the warning.

### Related information

* SonarC# Version 7.15.0.8572
* Visual Studio Version 16.1.4

**Comment by andrei-epure-sonarsource** [comment]
thanks @antoinebj  I confirm it's a FP

**Comment by stefanloerwald** [comment]
This is still an active issue now. Is there any plan to fix it?

**Comment by andrei-epure-sonarsource** [comment]
It's in our backlog, @stefanloerwald , so we will address it at some point in the future. I don't have an ETA for that, though.
